lib/serverengine/process_manager.rb

```diff
@@ -146,9 +151,19 @@ def spawn(*args)
       rpipe, wpipe = new_pipe_pair
 
       begin
-        options[[wpipe.fileno]] = wpipe
-        if @enable_heartbeat
-          env['SERVERENGINE_HEARTBEAT_PIPE'] = wpipe.fileno.to_s
+
+        if ServerEngine.windows?
+          @heartbeat_num += 1
+          pipe_name = "SERVERENGINE_HEARTBEAT_PIPE_" + @heartbeat_num.to_s
```

Comment:
If my understanding is correct, named pipes on Windows is in global namespace. So using incremental numbers locally (0..) conflicts with another Ruby process which uses ServerEngine. Correct? If yes, here needs to generate globally unique number on this machine.

An idea is like this:
```
pipe_name = "SERVERENGINE_HEARTBEAT_PIPE_%016x" % Random.new.rand(2**128)
```